test/python/test_tools_import_special_phrases.py

```diff
@@ -0,0 +1,120 @@
+"""
+    Tests for import special phrases functions
+"""
+from pathlib import Path
+import pytest
+from nominatim.tools.special_phrases import SpecialPhrasesImporter
+
+TEST_BASE_DIR = Path(__file__) / '..' / '..'
+
+def test_process_amenity_with_operator(special_phrases_importer, getorcreate_amenityoperator_funcs):
+    special_phrases_importer._process_amenity('', '', '', '', 'near')
+    special_phrases_importer._process_amenity('', '', '', '', 'in')
+
+def test_process_amenity_without_operator(special_phrases_importer, getorcreate_amenity_funcs):
+    special_phrases_importer._process_amenity('', '', '', '', '')
+
+def test_create_place_classtype_indexes(temp_db_conn, special_phrases_importer):
+    phrase_class = 'class'
+    phrase_type = 'type'
+    table_name = 'place_classtype_{}_{}'.format(phrase_class, phrase_type)
+    index_prefix = 'idx_place_classtype_{}_{}_'.format(phrase_class, phrase_type)
+
+    with temp_db_conn.cursor() as temp_db_cursor:
+        temp_db_cursor.execute("CREATE EXTENSION postgis;")
+        temp_db_cursor.execute('CREATE TABLE {}(place_id BIGINT, centroid GEOMETRY)'.format(table_name))
+
+    special_phrases_importer._create_place_classtype_indexes('', phrase_class, phrase_type)
+
+    centroid_index_exists = temp_db_conn.index_exists(index_prefix + 'centroid')
+    place_id_index_exists = temp_db_conn.index_exists(index_prefix + 'place_id')
+
+    assert centroid_index_exists and place_id_index_exists
+
+def test_create_place_classtype_table(temp_db_conn, placex_table, special_phrases_importer):
+    phrase_class = 'class'
+    phrase_type = 'type'
+    special_phrases_importer._create_place_classtype_table('', phrase_class, phrase_type)
+
+    with temp_db_conn.cursor() as temp_db_cursor:
```

Comment:
There is a 'temp_db_cursor' fixture for your convenience.

Reply:
At first I didn't use it because in my SpecialPhrasesImporter class I am only committing changes to the database when the whole import process is done (at the end of "import_from_wiki"), so if I use 2 different connection I get an AssertionError as the result is not found (as my class SpecialPhrasesImporter take in a db_connection and temp_db_cursor get a new one too).

Should I commit more often during the process and use this temp_db_cursor fixture?